File: Daily/sbgo/Leetcode137-只出现一次的数字.go
package sbgo

func singleNumber(nums []int) int {
	hash := map[int]int{}
	for _, num := range nums {
		hash[num]++
	}
	ans := 0
	for k, v := range hash {
		if v == 1 {
			ans = k
			break
		}
	}
	return ans
}

File: Daily/Rust/Leetcode137-只出现一次的数字.rs
use std::collections::HashMap;

impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut hash = HashMap::new();
        for num in nums{
            let count = hash.entry(num).or_insert(0);
            *count += 1;
        }
        let mut ans = 0;
        for (key,value) in &hash{
            if *value == 1{
                ans = *key;
                break;
            }
        }
        ans
    }
}